internal/exports/aws_s3_writer_test.go

```diff
@@ -263,6 +263,7 @@ func TestAWSS3CallbackFactory(t *testing.T) {
 			for _, record := range tt.records {
 				callback(record)
 			}
+			wg.Wait()
```

Comment:
this seems wrong in that it doesn't follow the logic in the implementation. the wait should be part of the teardown function.

Reply:
or it should never be part of any of the teardown functions, I suppose.

Reply:
Without it, the test causes a race, which I so far haven't been able to reproduce outside the test.

Reply:
I've pushed a new solution, but I'm not sure of the broader implications of the change...

Reply:
do we fail to pass the waitgroup by reference somewhere? because if it requires you to put it there on 266, then it is a problem in the actual code, not the test itself.

Reply:
I removed it from 266 and made a change in the `teardown` instead.

Reply:
(And yes, I suspect the problem is in the actual code, though at the moment I can't tell you why I suspect this. `Guru meditation error 0x2A: Not enough tea.`)